atomic_reactor/plugins/exit_koji_promote.py

```diff
@@ -451,6 +452,16 @@ def add_log_type(output):
                 },
             },
         })
+        annotations = self.workflow.build_result.annotations
+        if annotations and 'digests' in annotations:
+            digests = {}
+            digests_annotations = json.loads(annotations['digests'])
+            for digest_annotation in digests_annotations:
```

Comment:
Code duplication. Consider moving this for-loop to helper method.

Reply:
Moved to a separate method in `utils`